tests/quantization/compressed_tensors_integration/test_compressed_models.py

```diff
@@ -80,7 +80,9 @@ def _has_nested_attr(obj, attr_path):
                     if comp_decomp_obj is not None and hasattr(submodule, "weight"):
                         if "sparse-only" in uncompressed_model:
                             self.assertTrue(
-                                torch.equal(submodule.weight, comp_decomp_obj.weight),
+                                torch.equal(
+                                    submodule.weight.to(torch_device), comp_decomp_obj.weight.to(torch_device)
+                                ),
```

Comment:
You have seen that these 2 could be on different device?

Reply:
Yes, I run it on Intel XPU and got the device error.